packages/mobx/src/core/derivation.ts

```diff
@@ -166,10 +166,12 @@ export function checkIfStateReadsAreAllowed(observable: IObservable) {
  */
 export function trackDerivedFunction<T>(derivation: IDerivation, f: () => T, context: any) {
     const prevAllowStateReads = allowStateReadsStart(true)
-    // pre allocate array allocation + room for variation in deps
-    // array will be trimmed by bindDependencies
     changeDependenciesStateTo0(derivation)
-    derivation.newObserving_ = new Array(derivation.observing_.length + 100)
+    // Preallocate array; will be trimmed by bindDependencies.
+    derivation.newObserving_ = new Array(
+        // Reserve constant space for initial dependencies, dynamic space otherwise.
+        derivation.runId_ === 0 ? 100 : derivation.observing_.length
```

Comment:
Yeah I was thinking we could potentially make this configurable e.g. via `configure({ preallocationBufferSize: <number> })`. But I'd rather not do that in this PR...and tbh I'm not sure how good idea that is anyway.

I'll try to carve out some time to find out what the impact of various values would be on Canva. But I suspect the (much) bigger win lies in reducing the non-initial preallocation‚—i.e., what this PR does—because the initialization is a one-off whereas repeatedly reading lots of computeds during an interaction (e.g. dragging) can really add up quickly with 60 fps.